artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -709,30 +732,41 @@ public final void sendBuffer(ByteBuf buffer, int deliveryCount) {
     *
     * @return a Netty ByteBuf containing the encoded bytes of this Message instance.
     */
-   public ReadableBuffer getSendBuffer(int deliveryCount) {
+   public ReadableBuffer getSendBuffer(int deliveryCount, MessageReference reference) {
       ensureDataIsValid();
 
+      DeliveryAnnotations deliveryAnnotations;
+
+      if (reference != null && reference.getProtocolData() != null && reference.getProtocolData() instanceof DeliveryAnnotations) {
+         deliveryAnnotations = (DeliveryAnnotations) reference.getProtocolData();
```

Comment:
@gemmellr the deliveryAnnotationsForSendBuffer is a bit hacky to me.. it was done by EnMaas team...  

I intend this to be the new way to do it now, as I need the encoding as part of the reference, and not the message.

it was ok for what they needed it.. but on my case the annotations are coming from the reference (which actually represents the delivery).

Reply:
The result is nearly incomprehensible, I'd consider the combination of the two far hackier than what was present before. E,g, it appears to be using one field for checking encoding sizes and yet then going off and encoding the other. Elsewhere if both fields happen to be set one lot gets silently dropped. What results is a mess, and likely doesnt entirely work in my opinion.

Reply:
so, my only option is to break EnMass use and force them to move to the new API.

Reply:
I dont really see that needing to be the case. I think even if having both APIs exist, it shouldnt be so hard to determine what the code is actually doing, and it could behave nicer than it currently looks to (e.g merge annotations if the broker wants to add its own, rather than silently dropping the other ones it would have otherwise sent).

(I'm also not sure it was EnMasse that added this btw, but doesnt really matter who did)